docs/coding-guidelines/vectorization-guidelines.md

```diff
@@ -0,0 +1,1060 @@
+public static unsafe Vector128<byte> Narrow(Vector128<ushort> lower, Vector128<ushort> upper)
+public static unsafe Vector128<ushort> Narrow(Vector128<uint> lower, Vector128<uint> upper)
+public static unsafe Vector128<uint> Narrow(Vector128<ulong> lower, Vector128<ulong> upper)
+```
+
+In contrary to [Sse2.PackUnsignedSaturate](https://learn.microsoft.com/dotnet/api/system.runtime.intrinsics.x86.sse2.packunsignedsaturate) and [AdvSimd.Arm64.UnzipEven](https://learn.microsoft.com/dotnet/api/system.runtime.intrinsics.arm.advsimd.arm64.unzipeven), `Narrow` applies a mask via AND to cut anything above the max value of returned vector:
+
+
+```cs
+Vector256<ushort> ushortVector = Vector256.Create((ushort)300);
+Console.WriteLine(ushortVector);
+unchecked { Console.WriteLine((byte)300); }
+Console.WriteLine(300 & byte.MaxValue);
+Console.WriteLine(Vector128.Narrow(ushortVector.GetLower(), ushortVector.GetUpper()));
+
+if (Sse2.IsSupported)
+{
+    Console.WriteLine(Sse2.PackUnsignedSaturate(ushortVector.GetLower().AsInt16(), ushortVector.GetUpper().AsInt16()));
+}
+```
+
+```log
+<300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300, 300>
+44
+44
+<44, 44, 44, 44, 44, 44, 44, 44, 44, 44, 44, 44, 44, 44, 44, 44>
+<255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255>
+```
+
+### Shuffle
+
+`Shuffle` creates a new vector by selecting values from an input vector using a set of indices (values that represent indexes if the input vector).
+
+```cs
+public static Vector128<int> Shuffle(Vector128<int> vector, Vector128<int> indices)
+public static Vector128<uint> Shuffle(Vector128<uint> vector, Vector128<uint> indices)
+public static Vector128<float> Shuffle(Vector128<float> vector, Vector128<int> indices)
+public static Vector128<long> Shuffle(Vector128<long> vector, Vector128<long> indices)
+public static Vector128<ulong> Shuffle(Vector128<ulong> vector, Vector128<ulong> indices)
+public static Vector128<double> Shuffle(Vector128<double> vector, Vector128<long> indices)
+```
+
+It can be used for many things, including reversing the input:
+
+```cs
+Vector128<int> intVector = Vector128.Create(100, 200, 300, 400);
+Console.WriteLine(intVector);
+Console.WriteLine(Vector128.Shuffle(intVector, Vector128.Create(3, 2, 1, 0)));
+```
+
+```log
+<100, 200, 300, 400>
+<400, 300, 200, 100>
+```
+
+#### Vector256.Shuffle vs Avx2.Shuffle
+
+`Vector256.Shuffle` and `Avx2.Shuffle` are not identical.
+
+`Avx2.Shuffle` is effectively `2x128-bit ops` and so if we do `Vector256.Shuffle(value, Vector256.Create(0L, 1L, 0L, 1L))` it is going to think we want `value[0], value[1], value[0], value[1]`. Where-as `Avx2.Shuffle` treats this as `value[0], value[1], value[2], value[3]`.
+
+While `Vector256.Shuffle` treats it as a "single 256-bit vector" (rather than "2x128-bit vectors"). This was done for consistency and to better map to a cross-platform mentality where `AVX-512` and `SVE` all operate on "full width".
```

Comment:
```suggestion
`Vector256.Shuffle` and `Avx2.Shuffle` are not identical.

`Avx2.Shuffle` is effectively `2x128-bit ops` while `Vector256.Shuffle` treats it as a "single 256-bit vector" (rather than "2x128-bit vectors"). This was done for consistency and to better map to a cross-platform mentality where `AVX-512` and `SVE` all operate on "full width".
```
I would not write more here. Or give full examples with their logged output to show the difference.